app/src/main/java/com/github/se/studybuddies/database/RealtimeDatabaseConnection.kt

```diff
@@ -0,0 +1,299 @@
+    return ChatVal.GROUPS + "/$groupUID/" + ChatVal.MESSAGES
+  }
+
+  private fun getTopicMessagesPath(groupUID: String, topicUID: String): String {
+    return ChatVal.GROUPS + "/$topicUID/" + ChatVal.TOPICS + "/$groupUID/" + ChatVal.MESSAGES
+  }
+
+  private fun getPrivateMessagesPath(chatUID: String): String {
+    return ChatVal.DIRECT_MESSAGES + "/$chatUID/" + ChatVal.MESSAGES
+  }
+
+  private fun getPrivateChatMembersPath(chatUID: String): String {
+    return ChatVal.DIRECT_MESSAGES + "/$chatUID/" + ChatVal.MEMBERS
+  }
+
+  private fun saveMessage(path: String, data: Map<String, Any>) {
+    rtDb
+        .getReference(path)
+        .updateChildren(data)
+        .addOnSuccessListener { Log.d("MessageSend", "Message successfully written!") }
+        .addOnFailureListener { e -> Log.w("MessageSend", "Failed to write message.", e) }
+  }
+
+  fun sendMessage(
+      chatUID: String,
+      message: Message,
+      chatType: ChatType,
+      additionalUID: String = ""
+  ) {
+    val messagePath = getMessagePath(chatUID, chatType, additionalUID) + "/${message.uid}"
+
+    val messageData =
+        mutableMapOf(
+            MessageVal.SENDER_UID to message.sender.uid, MessageVal.TIMESTAMP to message.timestamp)
+    when (message) {
+      is Message.TextMessage -> {
+        messageData[MessageVal.TEXT] = message.text
+        messageData[MessageVal.TYPE] = MessageVal.TEXT
+        saveMessage(messagePath, messageData)
+      }
+      is Message.PhotoMessage -> {
+
+        StorageDatabaseConnection().uploadChatImage(message.uid, chatUID, message.photoUri) { uri ->
+          if (uri != null) {
+            Log.d("MyPrint", "Successfully uploaded photo with uri: $uri")
+            messageData[MessageVal.PHOTO] = uri.toString()
+            messageData[MessageVal.TYPE] = MessageVal.PHOTO
+            saveMessage(messagePath, messageData)
+          } else {
+            Log.d("MyPrint", "Failed to upload photo")
+          }
+        }
+      }
+      is Message.FileMessage -> {
+        StorageDatabaseConnection().uploadChatFile(message.uid, chatUID, message.fileUri) { uri ->
+          if (uri != null) {
+            Log.d("MyPrint", "Successfully uploaded file with uri: $uri")
+            messageData[MessageVal.FILE] = uri.toString()
+            messageData[MessageVal.FILE_NAME] = message.fileName
+            messageData[MessageVal.TYPE] = MessageVal.FILE
+            saveMessage(messagePath, messageData)
+          } else {
+            Log.d("MyPrint", "Failed to upload file")
+          }
+        }
+      }
+      is Message.LinkMessage -> {
+        messageData[MessageVal.LINK] = message.linkUri.toString()
+        messageData[MessageVal.LINK_NAME] = message.linkName
+        messageData[MessageVal.TYPE] = MessageVal.LINK
+        saveMessage(messagePath, messageData)
+      }
+      else -> {
+        Log.d("MyPrint", "Message type not recognized")
+      }
+    }
+  }
+
+  fun getMessages(
```

Comment:
it is a small critique you can maybe change the name of the function so that it is clear what they do for example name this one fetchChatMessages